Options/Applicative/Common.hs

```diff
@@ -102,14 +102,14 @@ argMatches opt arg = case opt of
     return result
   CmdReader _ f ->
     flip fmap (f arg) $ \subp -> StateT $ \args -> do
-      setContext (Just arg) subp
+      enterContext arg subp
```

Comment:
Wouldn't a bracket exit the context on failure as well though? The point is that on failure we still have it.